app/waf/src/controllers/adc.controller.ts

```diff
@@ -482,6 +480,10 @@ export class AdcController extends BaseController {
   }
 
   private async setupOn(adc: Adc, addon: AddonReqValues): Promise<void> {
+    // install DO
+    if (await this.adcStCtr.readyTo(AdcState.DOINSTALLED))
```

Comment:
seems to me it does not need to check twice here.  see line #462 and #484

Reply:
accept this and I will make the change accordingly